### ISSUE TITLE
Service entrypoint is too magical

### ISSUE THREAD

**Issue by matti** [kontena/kontena#2567]
If image has `ENTRYPOINT` defined, then kontena takes that and puts it to docker's `CMD` as `[imagesentrypoint, cmd1, cmd2, cmd3...]` (because entrypoint is reserved for weave)

user has no way to override / unset this entrypoint.

also, kontena service show doesn't show this magic:

`kontena service create myservice someimage/withentrypoint:latest` and then
`kontena service show myservice` will show:

```
 cmd: sleep 60
```

but `docker inspect myservice-1` shows what is really happening:

```
           "Cmd": [
                "/app/entry_point.sh",
                "sleep",
                "60"
            ],
```

maybe kontena service create/update should have `--entrypoint` option (currently has `--cmd`) that would prevent this magic from happening? kontena service show etc should separate `entrypoint:` even if it would internally put everything together as cmd.

**Comment by matti** [comment]
also what happens is that `/app/entry_point.sh` will get run over and over again as kontena service keeps restarting (by design) and user has no clue because it's not visible anywhere else than in `docker inspect`

**Comment by matti** [comment]
workaround (maybe) `net_mode: host`

**Comment by jakolehm** [comment]
One more reason to get rid of entrypoint hacks.

**Comment by jnummelin** [comment]
yes, definitely best way to avoid these kind of confusing things would be to get rid of the entrypoint hacks.

Getting the entrypoint info to `service show` etc. commands needs some mapping from agent side. Only agent will see the entrypoint after it creates the container or from the image info. As server does not know the image details, it needs agent to send the info back during deploy.
